### PR TITLE
Disable stack check on Macs/clang

### PR DESCRIPTION
**Pull request by mrallen1**
Based on discussion in https://github.com/asdf-vm/asdf-erlang/issues/116#issuecomment-539915106, do not munge CFLAGS on Catalina.

### CODE REVIEW

**Comment by feld** [comment]
are you hitting this? This what I'm seeing

```
Build failed.
 MAKE	emulator
 MAKE	opt
 GEN	x86_64-apple-darwin19.0.0/gen_git_version.mk
 PROFILE beam.prof.smp
/bin/sh: line 1: 64505 Segmentation fault: 11  LLVM_PROFILE_FILE="obj/x86_64-apple-darwin19.0.0/opt/smp/erlc-%m.profraw" ERL_FLAGS="-emu_type prof +S 1" erlc -W -DPGO -o obj/x86_64-apple-darwin19.0.0/opt/smp test/estone_SUITE.erl > obj/x86_64-apple-darwin19.0.0/opt/smp/PROFILE_LOG
make[4]: *** [obj/x86_64-apple-darwin19.0.0/opt/smp/PROFILE] Error 139
make[3]: *** [opt] Error 2
make[2]: *** [opt] Error 2
make[1]: *** [smp] Error 2
make: *** [emulator] Error 2
```

**Comment by mrallen1** [comment]
I don't have a system running Catalina yet but several other reports have come in about erlc giving a segfault during compilation.

**Comment by mrallen1** [comment]
https://github.com/erlang/otp/pull/2413

**Comment by mrleeio** [comment]
@mrallen1 first I want to thank you for attempting to fix the issue.

Sadly your change has not allowed me to build Erlang v19 on 10.15

```
~ % kerl build 19.3
Downloading otp_src_19.3.tar.gz to /Users/mrleeio/.kerl/archives
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   162  100   162    0     0    178      0 --:--:-- --:--:-- --:--:--   178
100 65.3M  100 65.3M    0     0   322k      0  0:03:27  0:03:27 --:--:--  359k
Getting checksum file from erlang.org...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   162  100   162    0     0    236      0 --:--:-- --:--:-- --:--:--   236
100 32419  100 32419    0     0  18192      0  0:00:01  0:00:01 --:--:-- 51605
Verifying archive checksum...
Checksum verified (a8c259ec47bf84e77510673e1b76b6db)
Extracting source code
Building Erlang/OTP 19.3 (19.3), please wait...
DOCUMENTATION INFORMATION (See: /Users/mrleeio/.kerl/builds/19.3/otp_build_19.3.log)
 * documentation  :
 *                  fop is missing.
 *                  Using fakefop to generate placeholder PDF files.

Build failed.
Makefile:72: warning: overriding commands for target `clean'
/Users/mrleeio/.kerl/builds/19.3/otp_src_19.3/make/otp_subdir.mk:29: warning: ignoring old commands for target `clean'
=== Entering application hipe
 VSN	hipe.hrl
 ERLC	../ebin/hipe_rtl.beam
erts_mmap: Failed to create super carrier of size 512 MB
make[3]: *** [../ebin/hipe_rtl.beam] Error 1
make[2]: *** [opt] Error 2
make[1]: *** [opt] Error 2
make: *** [secondary_bootstrap_build] Error 2

Please see /Users/mrleeio/.kerl/builds/19.3/otp_build_19.3.log for full details.
```

Open to other suggestions you might have and continued testing of this pull-request.

**Comment by rjcoelho** [comment]
Same error as @mrleeio

**Comment by mrleeio** [comment]
@rjcoelho found a Japanese blog entry which explores the issue in more depth.

https://qiita.com/irisTa56/items/21aa6dfe2402c6ba6abc

TL;DR this pull-request will only work for Erlang > v21, anything older which requires `hipe` will continue to fail.

Potential work around is to use the following kerl config:

```
KERL_CONFIGURE_OPTIONS="--disable-hipe --with-ssl=/usr/local/opt/openssl"
```

In combination with the stack-check disabled:

```
export CFLAGS="-O2 -g -fno-stack-check"
```

Will try it out when I get home.
